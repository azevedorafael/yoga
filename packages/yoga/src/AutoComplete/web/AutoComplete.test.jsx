import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { ThemeProvider, AutoComplete } from '../..';

describe('<AutoComplete />', () => {
  describe('Snapshots', () => {
    it('should match with default component', () => {
      const { container } = render(
        <ThemeProvider>
          <AutoComplete />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match with a full width component', () => {
      const { container } = render(
        <ThemeProvider>
          <AutoComplete full />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match with options list', () => {
      const { getByDisplayValue, container } = render(
        <ThemeProvider>
          <AutoComplete value="s" options={['first', 'second']} />
        </ThemeProvider>,
      );

      fireEvent.focus(getByDisplayValue('s'));

      expect(container).toMatchSnapshot();
    });

    it('should match with full width options list', () => {
      const { getByDisplayValue, container } = render(
        <ThemeProvider>
          <AutoComplete value="s" full options={['first', 'second']} />
        </ThemeProvider>,
      );

      fireEvent.focus(getByDisplayValue('s'));

      expect(container).toMatchSnapshot();
    });
  });

  describe('Options list', () => {
    it('should show the list when autocomplete has a value and focus', () => {
      const { getByDisplayValue, getByText } = render(
        <ThemeProvider>
          <AutoComplete value="New" options={['New York']} />
        </ThemeProvider>,
      );

      fireEvent.focus(getByDisplayValue('New'));
      expect(getByText('York')).toBeTruthy();
    });

    it('should filter the options', () => {
      const { container, getByDisplayValue } = render(
        <ThemeProvider>
          <AutoComplete value="second" options={['first', 'second', 'third']} />
        </ThemeProvider>,
      );

      fireEvent.focus(getByDisplayValue('second'));

      expect(container.querySelector('ul').firstChild.textContent).toBe(
        'second',
      );
    });

    it('should close options list when clean button is clicked', () => {
      const { container, getByDisplayValue, getByRole } = render(
        <ThemeProvider>
          <AutoComplete value="second" options={['first', 'second', 'third']} />
        </ThemeProvider>,
      );

      fireEvent.focus(getByDisplayValue('second'));

      fireEvent.click(getByRole('button'));

      expect(container.querySelector('ul')).toBe(null);
    });
  });

  describe('Event listeners', () => {
    it('should call onChangeMock', () => {
      const onChangeMock = jest.fn();

      const { getByDisplayValue } = render(
        <ThemeProvider>
          <AutoComplete
            value="New"
            options={['New York']}
            onChange={onChangeMock}
          />
        </ThemeProvider>,
      );

      fireEvent.change(getByDisplayValue('New'), { target: { value: 'a' } });

      expect(onChangeMock).toHaveBeenCalled();
    });

    it('should call onCleanMock', () => {
      const onCleanMock = jest.fn();

      const { getByDisplayValue, getByRole } = render(
        <ThemeProvider>
          <AutoComplete
            value="New"
            options={['New York']}
            onClean={onCleanMock}
          />
        </ThemeProvider>,
      );

      fireEvent.focus(getByDisplayValue('New'));
      fireEvent.click(getByRole('button'));

      expect(onCleanMock).toHaveBeenCalledWith('');
    });
  });

  describe('keyDown events', () => {
    it('should change selected item when arrow down key is pressed', () => {
      const { container, getByDisplayValue } = render(
        <ThemeProvider>
          <AutoComplete value="New" options={['New A', 'New B', 'New C']} />
        </ThemeProvider>,
      );

      fireEvent.focus(getByDisplayValue('New'));
      fireEvent.keyDown(getByDisplayValue('New'), { key: 'ArrowDown' });
      fireEvent.keyDown(getByDisplayValue('New'), { key: 'ArrowDown' });

      expect(document.activeElement).toBe(
        container.querySelector('ul').children[1],
      );
    });

    it('should change selected item when arrow up key is pressed', () => {
      const { container, getByDisplayValue } = render(
        <ThemeProvider>
          <AutoComplete value="New" options={['New A', 'New B', 'New C']} />
        </ThemeProvider>,
      );

      fireEvent.focus(getByDisplayValue('New'));
      fireEvent.keyDown(getByDisplayValue('New'), { key: 'ArrowDown' });
      fireEvent.keyDown(getByDisplayValue('New'), { key: 'ArrowDown' });
      fireEvent.keyDown(getByDisplayValue('New'), { key: 'ArrowUp' });

      expect(document.activeElement).toBe(
        container.querySelector('ul').children[0],
      );
    });
  });

  describe('onSelect event', () => {
    it('should select an option when click on it', () => {
      const onSelectMock = jest.fn();

      const { container, getByDisplayValue } = render(
        <ThemeProvider>
          <AutoComplete
            value="New"
            options={['New A', 'New B', 'New C']}
            onSelect={onSelectMock}
          />
        </ThemeProvider>,
      );

      fireEvent.focus(getByDisplayValue('New'));
      const { firstChild } = container.querySelector('ul');
      const selectedText = firstChild.textContent;

      fireEvent.click(firstChild);

      expect(onSelectMock).toHaveBeenCalledWith(selectedText);
    });

    it('should select an option when enter key is pressed on it', () => {
      const onSelectMock = jest.fn();

      const { getByDisplayValue } = render(
        <ThemeProvider>
          <AutoComplete
            value="New"
            options={['New A', 'New B', 'New C']}
            onSelect={onSelectMock}
          />
        </ThemeProvider>,
      );

      const input = getByDisplayValue('New');

      fireEvent.focus(input);
      fireEvent.keyDown(input, { key: 'ArrowDown' });

      const selectedOption = document.activeElement;

      fireEvent.keyDown(selectedOption, { key: 'Enter' });

      expect(onSelectMock).toHaveBeenCalledWith(selectedOption.textContent);
    });
  });
});
