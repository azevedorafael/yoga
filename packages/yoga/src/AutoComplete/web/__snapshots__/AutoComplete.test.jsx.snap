// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AutoComplete /> Snapshots should match with a full width component 1`] = `
.c2 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  height: 52px;
}

.c2 {
  width: 100%;
}

.c2 svg {
  position: absolute;
  top: 0;
  right: 0;
  padding-right: 16px;
  padding-left: 8px;
  height: 52px;
  width: 20px;
  fill: #9898A7;
  outline: none;
  box-sizing: content-box;
  cursor: pointer;
}

.c2 svg:hover,
.c2 svg:focus {
  fill: #41414A;
}

.c2:focus-within legend {
  max-width: 1000px;
  font-weight: 700;
}

.c2:hover fieldset,
.c2:focus-within fieldset {
  border-color: #41414A;
}

.c4 {
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  outline: none;
  height: 100%;
  padding-top: 16px;
  padding-right: 40px;
  padding-bottom: 16px;
  padding-left: 16px;
  color: #41414A;
  border: none;
  font-family: Open Sans,sans-serif;
  font-size: 14px;
  font-weight: 400;
  box-sizing: border-box;
}

.c4:focus {
  color: #41414A;
}

.c4:focus + label {
  top: 0;
  left: 16px;
  font-size: 12px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  font-weight: 700;
  color: #41414A;
}

.c4:disabled {
  cursor: not-allowed;
  color: #D3D3E2;
}

.c4::-webkit-input-placeholder {
  color: #9898A7;
}

.c4::-moz-placeholder {
  color: #9898A7;
}

.c4:-ms-input-placeholder {
  color: #9898A7;
}

.c4::placeholder {
  color: #9898A7;
}

.c4[type="number"]::-webkit-outer-spin-button,
.c4[type="number"]::-webkit-inner-spin-button {
  display: none;
}

.c4:invalid {
  box-shadow: none;
}

.c6 {
  position: absolute;
  top: -7px;
  left: 0;
  padding: 0;
  padding-left: 0px;
  right: 0;
  bottom: 0;
  margin: 0;
  pointer-events: none;
  border-radius: 8px;
  border: 1px solid #D3D3E2;
  padding-left: 12px;
}

.c7 {
  position: relative;
  padding: 0;
  max-width: 0.01px;
  width: auto;
  height: 14px;
  font-weight: normal;
  -webkit-transition: max-width ease 50ms;
  transition: max-width ease 50ms;
  visibility: hidden;
  max-width: 0.01px;
  font-size: 12px;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c8 {
  padding-left: 4px;
  padding-right: 4px;
}

.c5 {
  position: absolute;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  top: 24px;
  left: 16px;
  font-size: 14px;
  font-weight: 400;
  color: #9898A7;
  -webkit-transform: translateY(-40%);
  -ms-transform: translateY(-40%);
  transform: translateY(-40%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
}

.c1 {
  width: 100%;
}

.c3 {
  z-index: 1;
}

.c0 {
  position: relative;
  width: 100%;
}

<div>
  <div
    aria-expanded="false"
    aria-haspopup="listbox"
    aria-labelledby="downshift-1-label"
    class="c0"
    role="combobox"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
        label="Label"
      >
        <input
          aria-autocomplete="list"
          aria-labelledby="downshift-1-label"
          autocomplete="off"
          class="c4"
          id="downshift-1-input"
          value=""
        />
        <label
          class="c5"
        >
          Label
        </label>
        <fieldset
          class="c6"
        >
          <legend
            class="c7"
          >
            <span
              class="c8"
            >
              Label
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AutoComplete /> Snapshots should match with default component 1`] = `
.c2 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  height: 52px;
}

.c2 {
  width: 320px;
}

.c2 svg {
  position: absolute;
  top: 0;
  right: 0;
  padding-right: 16px;
  padding-left: 8px;
  height: 52px;
  width: 20px;
  fill: #9898A7;
  outline: none;
  box-sizing: content-box;
  cursor: pointer;
}

.c2 svg:hover,
.c2 svg:focus {
  fill: #41414A;
}

.c2:focus-within legend {
  max-width: 1000px;
  font-weight: 700;
}

.c2:hover fieldset,
.c2:focus-within fieldset {
  border-color: #41414A;
}

.c4 {
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  outline: none;
  height: 100%;
  padding-top: 16px;
  padding-right: 40px;
  padding-bottom: 16px;
  padding-left: 16px;
  color: #41414A;
  border: none;
  font-family: Open Sans,sans-serif;
  font-size: 14px;
  font-weight: 400;
  box-sizing: border-box;
}

.c4:focus {
  color: #41414A;
}

.c4:focus + label {
  top: 0;
  left: 16px;
  font-size: 12px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  font-weight: 700;
  color: #41414A;
}

.c4:disabled {
  cursor: not-allowed;
  color: #D3D3E2;
}

.c4::-webkit-input-placeholder {
  color: #9898A7;
}

.c4::-moz-placeholder {
  color: #9898A7;
}

.c4:-ms-input-placeholder {
  color: #9898A7;
}

.c4::placeholder {
  color: #9898A7;
}

.c4[type="number"]::-webkit-outer-spin-button,
.c4[type="number"]::-webkit-inner-spin-button {
  display: none;
}

.c4:invalid {
  box-shadow: none;
}

.c6 {
  position: absolute;
  top: -7px;
  left: 0;
  padding: 0;
  padding-left: 0px;
  right: 0;
  bottom: 0;
  margin: 0;
  pointer-events: none;
  border-radius: 8px;
  border: 1px solid #D3D3E2;
  padding-left: 12px;
}

.c7 {
  position: relative;
  padding: 0;
  max-width: 0.01px;
  width: auto;
  height: 14px;
  font-weight: normal;
  -webkit-transition: max-width ease 50ms;
  transition: max-width ease 50ms;
  visibility: hidden;
  max-width: 0.01px;
  font-size: 12px;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c8 {
  padding-left: 4px;
  padding-right: 4px;
}

.c5 {
  position: absolute;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  top: 24px;
  left: 16px;
  font-size: 14px;
  font-weight: 400;
  color: #9898A7;
  -webkit-transform: translateY(-40%);
  -ms-transform: translateY(-40%);
  transform: translateY(-40%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
}

.c1 {
  width: auto;
}

.c3 {
  z-index: 1;
}

.c0 {
  position: relative;
  width: auto;
}

<div>
  <div
    aria-expanded="false"
    aria-haspopup="listbox"
    aria-labelledby="downshift-0-label"
    class="c0"
    role="combobox"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
        label="Label"
      >
        <input
          aria-autocomplete="list"
          aria-labelledby="downshift-0-label"
          autocomplete="off"
          class="c4"
          id="downshift-0-input"
          value=""
        />
        <label
          class="c5"
        >
          Label
        </label>
        <fieldset
          class="c6"
        >
          <legend
            class="c7"
          >
            <span
              class="c8"
            >
              Label
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AutoComplete /> Snapshots should match with full width options list 1`] = `
.c2 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  height: 52px;
}

.c2 {
  width: 100%;
}

.c2 svg {
  position: absolute;
  top: 0;
  right: 0;
  padding-right: 16px;
  padding-left: 8px;
  height: 52px;
  width: 20px;
  fill: #9898A7;
  outline: none;
  box-sizing: content-box;
  cursor: pointer;
}

.c2 svg:hover,
.c2 svg:focus {
  fill: #41414A;
}

.c2:focus-within legend {
  max-width: 1000px;
  font-weight: 700;
}

.c2:hover fieldset,
.c2:focus-within fieldset {
  border-color: #41414A;
}

.c4 {
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  outline: none;
  height: 100%;
  padding-top: 16px;
  padding-right: 40px;
  padding-bottom: 16px;
  padding-left: 16px;
  color: #41414A;
  border: none;
  font-family: Open Sans,sans-serif;
  font-size: 14px;
  font-weight: 400;
  box-sizing: border-box;
}

.c4:focus {
  color: #41414A;
}

.c4:focus + label {
  top: 0;
  left: 16px;
  font-size: 12px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  font-weight: 700;
  color: #41414A;
}

.c4:disabled {
  cursor: not-allowed;
  color: #D3D3E2;
}

.c4::-webkit-input-placeholder {
  color: #9898A7;
}

.c4::-moz-placeholder {
  color: #9898A7;
}

.c4:-ms-input-placeholder {
  color: #9898A7;
}

.c4::placeholder {
  color: #9898A7;
}

.c4 + label {
  top: 0;
  left: 16px;
  font-size: 12px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
}

.c4[type="number"]::-webkit-outer-spin-button,
.c4[type="number"]::-webkit-inner-spin-button {
  display: none;
}

.c4:invalid {
  box-shadow: none;
}

.c6 {
  position: absolute;
  top: -7px;
  left: 0;
  padding: 0;
  padding-left: 0px;
  right: 0;
  bottom: 0;
  margin: 0;
  pointer-events: none;
  border-radius: 8px;
  border: 1px solid #D3D3E2;
  padding-left: 12px;
}

.c7 {
  position: relative;
  padding: 0;
  max-width: 0.01px;
  width: auto;
  height: 14px;
  font-weight: normal;
  -webkit-transition: max-width ease 50ms;
  transition: max-width ease 50ms;
  visibility: hidden;
  max-width: 1000px;
  font-size: 12px;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c8 {
  padding-left: 4px;
  padding-right: 4px;
}

.c5 {
  position: absolute;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  top: 24px;
  left: 16px;
  font-size: 14px;
  font-weight: 400;
  color: #9898A7;
  -webkit-transform: translateY(-40%);
  -ms-transform: translateY(-40%);
  transform: translateY(-40%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
}

.c1 {
  width: 100%;
}

.c3 {
  z-index: 1;
  border-color: #41414A;
  border-bottom-width: 0;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.c3 fieldset {
  border-color: #41414A;
  border-bottom-width: 0;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.c3 label {
  color: #41414A;
  font-weight: 700;
}

.c3 legend {
  font-weight: 700;
}

.c0 {
  position: relative;
  width: 100%;
}

.c9 {
  position: absolute;
  margin: 0;
  padding: 0;
  z-index: 999;
  top: 52px;
  width: 100%;
  max-height: 312px;
  background-color: #FFFFFF;
  border-width: 1px;
  border-style: solid;
  border-color: #41414A;
  border-top-width: 0;
  border-bottom-left-radius: 8px;
  border-bottom-right-radius: 8px;
  overflow: hidden;
}

.c10 {
  list-style: none;
  cursor: pointer;
  padding: 16px 16px 16px 16px;
  background-color: #FFFFFF;
  outline: none;
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
}

.c10:hover {
  background-color: #F5F5FA;
}

.c11 {
  background: transparent;
  color: inherit;
  font-weight: 700;
}

<div>
  <div
    aria-expanded="true"
    aria-haspopup="listbox"
    aria-labelledby="downshift-3-label"
    aria-owns="downshift-3-menu"
    class="c0"
    role="combobox"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
        label="Label"
      >
        <input
          aria-autocomplete="list"
          aria-controls="downshift-3-menu"
          aria-labelledby="downshift-3-label"
          autocomplete="off"
          class="c4"
          id="downshift-3-input"
          value="s"
        />
        <label
          class="c5"
        >
          Label
        </label>
        <fieldset
          class="c6"
        >
          <legend
            class="c7"
          >
            <span
              class="c8"
            >
              Label
            </span>
          </legend>
        </fieldset>
        <svg
          role="button"
          tabindex="0"
        />
      </div>
    </div>
    <ul
      aria-labelledby="downshift-3-label"
      class="c9"
      id="downshift-3-menu"
      role="listbox"
    >
      <li
        aria-selected="false"
        class="c10"
        id="downshift-3-item-0"
        role="option"
      >
        
        <mark
          class="c11"
        >
          s
        </mark>
        econd
      </li>
      <li
        aria-selected="false"
        class="c10"
        id="downshift-3-item-1"
        role="option"
      >
        fir
        <mark
          class="c11"
        >
          s
        </mark>
        t
      </li>
    </ul>
  </div>
</div>
`;

exports[`<AutoComplete /> Snapshots should match with options list 1`] = `
.c2 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  height: 52px;
}

.c2 {
  width: 320px;
}

.c2 svg {
  position: absolute;
  top: 0;
  right: 0;
  padding-right: 16px;
  padding-left: 8px;
  height: 52px;
  width: 20px;
  fill: #9898A7;
  outline: none;
  box-sizing: content-box;
  cursor: pointer;
}

.c2 svg:hover,
.c2 svg:focus {
  fill: #41414A;
}

.c2:focus-within legend {
  max-width: 1000px;
  font-weight: 700;
}

.c2:hover fieldset,
.c2:focus-within fieldset {
  border-color: #41414A;
}

.c4 {
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  outline: none;
  height: 100%;
  padding-top: 16px;
  padding-right: 40px;
  padding-bottom: 16px;
  padding-left: 16px;
  color: #41414A;
  border: none;
  font-family: Open Sans,sans-serif;
  font-size: 14px;
  font-weight: 400;
  box-sizing: border-box;
}

.c4:focus {
  color: #41414A;
}

.c4:focus + label {
  top: 0;
  left: 16px;
  font-size: 12px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  font-weight: 700;
  color: #41414A;
}

.c4:disabled {
  cursor: not-allowed;
  color: #D3D3E2;
}

.c4::-webkit-input-placeholder {
  color: #9898A7;
}

.c4::-moz-placeholder {
  color: #9898A7;
}

.c4:-ms-input-placeholder {
  color: #9898A7;
}

.c4::placeholder {
  color: #9898A7;
}

.c4 + label {
  top: 0;
  left: 16px;
  font-size: 12px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
}

.c4[type="number"]::-webkit-outer-spin-button,
.c4[type="number"]::-webkit-inner-spin-button {
  display: none;
}

.c4:invalid {
  box-shadow: none;
}

.c6 {
  position: absolute;
  top: -7px;
  left: 0;
  padding: 0;
  padding-left: 0px;
  right: 0;
  bottom: 0;
  margin: 0;
  pointer-events: none;
  border-radius: 8px;
  border: 1px solid #D3D3E2;
  padding-left: 12px;
}

.c7 {
  position: relative;
  padding: 0;
  max-width: 0.01px;
  width: auto;
  height: 14px;
  font-weight: normal;
  -webkit-transition: max-width ease 50ms;
  transition: max-width ease 50ms;
  visibility: hidden;
  max-width: 1000px;
  font-size: 12px;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c8 {
  padding-left: 4px;
  padding-right: 4px;
}

.c5 {
  position: absolute;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  top: 24px;
  left: 16px;
  font-size: 14px;
  font-weight: 400;
  color: #9898A7;
  -webkit-transform: translateY(-40%);
  -ms-transform: translateY(-40%);
  transform: translateY(-40%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
}

.c1 {
  width: auto;
}

.c3 {
  z-index: 1;
  border-color: #41414A;
  border-bottom-width: 0;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.c3 fieldset {
  border-color: #41414A;
  border-bottom-width: 0;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.c3 label {
  color: #41414A;
  font-weight: 700;
}

.c3 legend {
  font-weight: 700;
}

.c0 {
  position: relative;
  width: auto;
}

.c9 {
  position: absolute;
  margin: 0;
  padding: 0;
  z-index: 999;
  top: 52px;
  width: 320px;
  max-height: 312px;
  background-color: #FFFFFF;
  border-width: 1px;
  border-style: solid;
  border-color: #41414A;
  border-top-width: 0;
  border-bottom-left-radius: 8px;
  border-bottom-right-radius: 8px;
  overflow: hidden;
}

.c10 {
  list-style: none;
  cursor: pointer;
  padding: 16px 16px 16px 16px;
  background-color: #FFFFFF;
  outline: none;
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
}

.c10:hover {
  background-color: #F5F5FA;
}

.c11 {
  background: transparent;
  color: inherit;
  font-weight: 700;
}

<div>
  <div
    aria-expanded="true"
    aria-haspopup="listbox"
    aria-labelledby="downshift-2-label"
    aria-owns="downshift-2-menu"
    class="c0"
    role="combobox"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
        label="Label"
      >
        <input
          aria-autocomplete="list"
          aria-controls="downshift-2-menu"
          aria-labelledby="downshift-2-label"
          autocomplete="off"
          class="c4"
          id="downshift-2-input"
          value="s"
        />
        <label
          class="c5"
        >
          Label
        </label>
        <fieldset
          class="c6"
        >
          <legend
            class="c7"
          >
            <span
              class="c8"
            >
              Label
            </span>
          </legend>
        </fieldset>
        <svg
          role="button"
          tabindex="0"
        />
      </div>
    </div>
    <ul
      aria-labelledby="downshift-2-label"
      class="c9"
      id="downshift-2-menu"
      role="listbox"
    >
      <li
        aria-selected="false"
        class="c10"
        id="downshift-2-item-0"
        role="option"
      >
        
        <mark
          class="c11"
        >
          s
        </mark>
        econd
      </li>
      <li
        aria-selected="false"
        class="c10"
        id="downshift-2-item-1"
        role="option"
      >
        fir
        <mark
          class="c11"
        >
          s
        </mark>
        t
      </li>
    </ul>
  </div>
</div>
`;
